test/e2e/framework/job/fixtures.go

```diff
@@ -126,6 +129,7 @@ func NewTestJobOnNode(behavior, name string, rPol v1.RestartPolicy, parallelism,
 		// the non-0-indexed pods are succeeded is used to determine that the
 		// 0th indexed pod already created the marker file.
 		setupHostPathDirectory(job)
+		job.Spec.Template.Spec.TerminationGracePeriodSeconds = ptr.To(int64(1))
```

Comment:
Does this have the same problem?
sleep wouldn't be PID 1

Reply:
I have verified experimentally that it helps to reduce the time of the "Using a pod failure policy to not count some failures towards the backoffLimit" tests by around 29s. Previously, the tests are waiting until SIGKILL to kill the pod (thus the 137 exit code). SIGTERM does not kill the pod because the sleep is actually called directly from the entrypoint script, which is PID1 (or at least this is how my understanding goes).